sa/storage-authority.go

```diff
@@ -166,6 +166,20 @@ func (ssa *SQLStorageAuthority) GetAuthorization(id string) (authz core.Authoriz
 	return
 }
 
+// Get the valid authorization with biggest expire date for a given domain and registrationId
+func (ssa *SQLStorageAuthority) GetLatestValidAuthorization(registrationId int64, identifier core.AcmeIdentifier) (authz core.Authorization, err error) {
+	ident, err := json.Marshal(identifier)
```

Comment:
I'm not sure I understand your comment ? I also need to check the identifier Type.
Note that I don't think the current database format is optimal. I don't like storing json data in database. Right now the only way of looking for an authorization for a given ident is too Marshal the AcmeIdentifier, and perform a string lookup in the db. If for any reason the Marshal function changes (for instance if the json attributes are not encoded in the same order), it will stop working.
There is the same kind of problem for looking up registration by key.